### ISSUE TITLE
cost control - what happens when multiple jobs run on the same instance

### ISSUE THREAD

**Issue by rvencu** [aws-samples/pcluster-manager#267]
We share some instances for multiple users, and I see the prolog and epilog scripts are adding/removing info from the job files while the crontab script is taking care of maintaining the tags at the instance level.

What happens in the billing section when multiple jobs run in parallel on the same instance?
1. will be the tag value like project1 project2 ... ?
2. how about the dollar figure, is it split among listed projects and eventually idle resources, or is it duplicated in full among all projects?

**Comment by mendaomn** [comment]
We're looking into the matter, will come back to you with some insights

**Comment by mendaomn** [comment]
According to my research, you will have to look at the cost of cluster using the [CUR data](https://docs.aws.amazon.com/cur/latest/userguide/what-is-cur.html), and then allocate the cost per job using Slurm account based on the compute hours. 

This [notebook](https://github.com/aws-samples/aws-research-workshops/blob/mainline/notebooks/parallelcluster/pcluster-accounting.ipynb) by [@jxuamazon](https://github.com/jxuamazon) outlines how to do it. 

Please not the allocation is not 100% accurate, as shared resources (head node, compute node scale-up/downs) costs are approximated

Hope this helps

**Comment by sean-smith** [comment]
@rvencu, I assume you're referring to the [Cost Tracking](https://pcluster.cloud/02-tutorials/04-cost-tracking.html) guide. In that guide we make the assumption that each job is filling full instances, i.e. using `--exclusive` flag. If this isn't the case you can use Slurm accounting to get instance hours then account for job cost based on number of CPU (or GPU) hours consumed.

Another approach is to use smaller instances based on the size of the job, with exception of a few instance types (p4d), you can typically size the instance to the job, i.e. if I had a 18 core job I'd use c5n.9xlarge vs. the full 36 core option on c5n.18xlarge. This makes the cost-tracking process easier as you can get accurate costs through cost-explorer.

**Comment by mendaomn** [comment]
Closing the issue. Feel free to open another issue if you have further questions!
